Build failure: ahk_x11

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix build nixpkgs#ahk_x11`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310520474

### Relevant log output

```console
last 25 log lines:
       > warn - atspi_accessible_get_attributes - Unknown conversion to crystal for GHash
       > warn - atspi_document_get_attributes - Unknown conversion to crystal for GHash
       > warn - atspi_text_get_attribute_run - Unknown conversion to crystal for GHash
       > warn - atspi_text_get_attributes - Unknown conversion to crystal for GHash
       > warn - atspi_text_get_default_attributes - Unknown conversion to crystal for GHash
       > warn - g_hash_table_ref - Unknown conversion to crystal for GHash
       > info - Gio - No binding config found for GModule-2.0.
       > warn - g_cancellable_connect - Callback without user_data!
       > warn - Gio::ActionEntry padding field - Unknown conversion to crystal for fixed size array.
       > warn - Gio::DBusInterfaceVTable padding field - Unknown conversion to crystal for fixed size array.
       > warn - Gio::DBusSubtreeVTable padding field - Unknown conversion to crystal for fixed size array.
       > info - Gdk - No binding config found for cairo-1.0.
       > info - Gdk - No binding config found for GdkPixbuf-2.0.
       > warn - Gdk - WMDecoration::All (0x1) doesn't have all possible bits set (0x7f).
       > warn - Gdk - WMFunction::All (0x1) doesn't have all possible bits set (0x3f).
       > info - HarfBuzz - No binding config found for freetype2-2.0.
       > warn - HarfBuzz - Interface constant not supported.
       > warn - gdk_pixbuf_get_options - Unknown conversion to crystal for GHash
       > info - Gtk - No binding config found for Atk-1.0.
       > warn - Settings#color-hash - Unknown conversion to crystal for GHash
       > syntax error in './harf_buzz-0.0/harf_buzz.cr:7846:20': can't use variable name 'glyphs_before' inside assignment to variable 'glyphs_before'
       > fatal - Error formating generated files at '/build/source/lib/gi-crystal/src/auto'.
       > make[1]: *** [Makefile:63: lib/configured] Error 1
       > make[1]: Leaving directory '/build/source'
       > make: *** [Makefile:32: bin/ahk_x11] Error 2
```

### Additional context

_No response_

### System metadata

fetching path input 'path:/nix/store/jd3y5kfy3ywv9lc229nz8vxqbrcyv93d-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.5, NixOS, 25.11 (Xantusia), 25.11.20251025.6a08e6b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/user/.config/nix/nix.conf:/nix/store/60s7z65sdmfnjmyfbkznhs12c7xcsqkn-plasma-workspace-6.5.0/etc/xdg/nix/nix.conf:/nix/store/w7ijd1nbilqf7818rwkdq4y2gdyx17ri-kglobalacceld-6.5.0/etc/xdg/nix/nix.conf:/nix/store/kkspq3hkpk1pma05268yxz5cpd00fvpr-baloo-6.19.0/etc/xdg/nix/nix.conf:/home/user/.config/kdedefaults/nix/nix.conf:/nix/store/60s7z65sdmfnjmyfbkznhs12c7xcsqkn-plasma-workspace-6.5.0/etc/xdg/nix/nix.conf:/nix/store/w7ijd1nbilqf7818rwkdq4y2gdyx17ri-kglobalacceld-6.5.0/etc/xdg/nix/nix.conf:/nix/store/kkspq3hkpk1pma05268yxz5cpd00fvpr-baloo-6.19.0/etc/xdg/nix/nix.conf:/home/user/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/user/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/user/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/user/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/user/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/sja6jggc51h6cgq6iqffrjykh2lx6hw1-lix-2.93.3/share`
 - nixpkgs: `/nix/store/jd3y5kfy3ywv9lc229nz8vxqbrcyv93d-source`


### Notify maintainers

@UlyssesZh

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ah/ahk_x11/package.nix
+++ b/pkgs/by-name/ah/ahk_x11/package.nix
@@ -20,7 +20,7 @@
 # NOTICE: AHK_X11 from this package does not support compiling scripts into portable executables.
 let
   pname = "ahk_x11";
-  version = "1.0.6";
+  version = "1.0.7";
 
   inherit (xorg)
     libXinerama
@@ -37,7 +37,7 @@ crystal.buildCrystalPackage {
     owner = "phil294";
     repo = "AHK_X11";
     tag = version;
-    hash = "sha256-t2fGUIG3T8azx22lFhFAkABHwkePv9uThhlH+fwDj8E=";
+    hash = "sha256-VuqLlRgNyF6/4aVq1sNlVjOMih0TdHXbr0CqhA4QT6Y=";
     fetchSubmodules = true;
   };
 

--- a/pkgs/by-name/ah/ahk_x11/shards.nix
+++ b/pkgs/by-name/ah/ahk_x11/shards.nix
@@ -1,60 +1,60 @@
 {
-  atspi = {
+  "atspi" = {
     url = "https://github.com/phil294/atspi.cr.git";
     rev = "7aec8d3604313da415e54e654c1dc6b33ae05d8d";
     sha256 = "1r6k9f00a2y406hqsgq0r58zlfmg3xbzqjmfh93nwqs4m6h0shml";
   };
-  cron_parser = {
+  "cron_parser" = {
     url = "https://github.com/kostya/cron_parser.git";
     rev = "v0.4.0";
     sha256 = "17fgg2nvyx99v05l10h6cnxfr7swz8yaxhmnk4l47kg2spi8w90a";
   };
-  future = {
+  "future" = {
     url = "https://github.com/crystal-community/future.cr.git";
     rev = "v1.0.0";
     sha256 = "1mji2djkrf4vxgs432kgkzxx54ybzk636789k2vsws3sf14l74i8";
   };
-  gi-crystal = {
+  "gi-crystal" = {
     url = "https://github.com/hugopl/gi-crystal.git";
     rev = "v0.23.0";
     sha256 = "0ykb7sy62036brix12sj416h6gx85liq2kivwkh2cmf71lk02cqr";
   };
-  gtk3 = {
+  "gtk3" = {
     url = "https://github.com/phil294/gtk3.cr.git";
     rev = "3e6cdc45c1c39d82c215b4cdf0bbb9b5598460e1";
     sha256 = "0x25p85kp2f6gbsi36nyddhji7mmmqa7v4xs18bddf3kbgh72nw3";
   };
-  harfbuzz = {
+  "harfbuzz" = {
     url = "https://github.com/hugopl/harfbuzz.cr.git";
-    rev = "v0.2.0";
-    sha256 = "06wgqxwyib5416yp53j2iwcbr3bl4jjxb1flm7z103l365par694";
+    rev = "v0.2.1";
+    sha256 = "08zx5q1mm1h4lzfr38dk8nk7mx1y955zdjl6n48rl619bc1ywdpg";
   };
-  notify = {
+  "notify" = {
     url = "https://github.com/phil294/notify.cr.git";
     rev = "0c9880e6169e09f0db1cb40432e0b76651d15ad3";
     sha256 = "1waasvhx4n817d8gqk3fza4ac4a9d71f80cfii5dbh2aqzhqf070";
   };
-  pango = {
+  "pango" = {
     url = "https://github.com/hugopl/pango.cr.git";
     rev = "v0.3.1";
     sha256 = "0xlf127flimnll875mcq92q7xsi975rrgdpcpmnrwllhdhfx9qmv";
   };
-  tasker = {
+  "tasker" = {
     url = "https://github.com/spider-gazelle/tasker.git";
     rev = "v2.1.4";
     sha256 = "0254sl279nrw5nz43dz5gm89ah1zrw5bvxfma81navpx5gfg9pyb";
   };
-  x11 = {
+  "x11" = {
     url = "https://github.com/phil294/x11-cr.git";
     rev = "5a937d647d8eac29106602fcdeb161a78eb3d59d";
     sha256 = "11rflkxrg1dl5c80llwan4qs47ir7w7lgqdmhy2z907kxbrfkznr";
   };
-  x_do = {
+  "x_do" = {
     url = "https://github.com/phil294/x_do.cr.git";
     rev = "94f755a4f18a9a9820e9738b371cbe6b49731926";
     sha256 = "099mrhyb8kly6isk7xr1mzxbxfm4xizmyyw2ylmlx4ywynz9iyl6";
   };
-  xtst = {
+  "xtst" = {
     url = "https://github.com/phil294/xtst-cr.git";
     rev = "f731b708e1789d039258ecd62f7b0d1107a00b78";
     sha256 = "1vhzn1ck7ifkvhw5f9kpljsa278va039wa6f3hhd8w6gr2zqvzk8";

